CMakePresets.json

```diff
@@ -8,93 +8,37 @@
-      "description": "obs-studio for macOS (Apple) - Release Configuration",
-      "inherits": ["macos-arm64"],
-      "cacheVariables": {
-        "ENABLE_RELEASE_BUILD": true,
-        "ENABLE_SPARKLE": true,
-        "OBS_SPARKLE_KEY": {"type": "STRING", "value": ""},
-        "OBS_CODESIGN_IDENTITY": "$penv{CODESIGN_IDENT}",
-        "OBS_CODESIGN_TEAM": "$penv{CODESIGN_TEAM}",
-        "SPARKLE_APPCAST_URL": {"type": "STRING", "value": "https://obsproject.com/osx_update/updates_arm64_v2.xml"},
-        "SPARKLE_PUBLIC_KEY": {"type": "STRING", "value": "HQ5/Ba9VHOuEWaM0jtVjZzgHKFJX9YTl+HNVpgNF0iM="},
+        "CMAKE_OSX_DEPLOYMENT_TARGET": {"type": "STRING", "value": "11.0"},
+        "OBS_CMAKE_VERSION": {"type": "STRING", "value": "3.0.0"},
+        "OBS_CODESIGN_TEAM": {"type": "STRING", "value": "$penv{CODESIGN_TEAM}"},
+        "OBS_CODESIGN_IDENTITY": {"type": "STRING", "value": "$penv{CODESIGN_IDENT}"},
+        "OBS_PROVISIONING_PROFILE": {"type": "STRING", "value": "$penv{PROVISIONING_PROFILE}"},
         "VIRTUALCAM_DEVICE_UUID": {"type": "STRING", "value": "7626645E-4425-469E-9D8B-97E0FA59AC75"},
         "VIRTUALCAM_SOURCE_UUID": {"type": "STRING", "value": "A8D7B8AA-65AD-4D21-9C42-66480DBFA8E1"},
-        "VIRTUALCAM_SINK_UUID": {"type": "STRING", "value": "A3F16177-7044-4DD8-B900-72E2419F7A9A"}
+        "VIRTUALCAM_SINK_UUID": {"type": "STRING", "value": "A3F16177-7044-4DD8-B900-72E2419F7A9A"},
+        "SPARKLE_APPCAST_URL": {"type": "STRING", "value": "https://obsproject.com/osx_update/updates_$(ARCHS)_v2.xml"},
+        "SPARKLE_PUBLIC_KEY": {"type": "STRING", "value": "HQ5/Ba9VHOuEWaM0jtVjZzgHKFJX9YTl+HNVpgNF0iM="}
       }
     },
     {
-      "name": "macos-release-x86_64",
-      "displayName": "macOS Intel (Release)",
-      "description": "obs-studio for macOS (Intel) - Release Configuration",
-      "inherits": ["macos-x86_64"],
+      "name": "macos-ci",
+      "displayName": "macOS (CI)",
+      "description": "CI macOS build (single architecture only)",
+      "inherits": ["macos"],
+      "warnings": {"dev": true, "deprecated": true},
       "cacheVariables": {
-        "ENABLE_RELEASE_BUILD": true,
-        "ENABLE_SPARKLE": true,
-        "OBS_SPARKLE_KEY": {"type": "STRING", "value": ""},
-        "OBS_CODESIGN_IDENTITY": "$penv{CODESIGN_IDENT}",
-        "OBS_CODESIGN_TEAM": "$penv{CODESIGN_TEAM}",
-        "SPARKLE_APPCAST_URL": {"type": "STRING", "value": "https://obsproject.com/osx_update/updates_x86_64_v2.xml"},
-        "SPARKLE_PUBLIC_KEY": {"type": "STRING", "value": "HQ5/Ba9VHOuEWaM0jtVjZzgHKFJX9YTl+HNVpgNF0iM="},
-        "VIRTUALCAM_DEVICE_UUID": {"type": "STRING", "value": "7626645E-4425-469E-9D8B-97E0FA59AC75"},
-        "VIRTUALCAM_SOURCE_UUID": {"type": "STRING", "value": "A8D7B8AA-65AD-4D21-9C42-66480DBFA8E1"},
-        "VIRTUALCAM_SINK_UUID": {"type": "STRING", "value": "A3F16177-7044-4DD8-B900-72E2419F7A9A"}
+        "CMAKE_COMPILE_WARNING_AS_ERROR": true
```

Comment:
Since this is enabled by default this seems redundant. (Or is the idea to only have it enabled on CI? In that case it'd probably need to be set to off either in the `macos` preset or by default)

Reply:
It is only enabled by default if the cache value is not set at all (because the CMake code does not set `-Werror` anymore and relies entirely on the cache variable).

Thus it is more correct to explicitly enable this in the preset (as the default value used by CMake is `FALSE`). I've actually came around to the opinion that enforcing this actually violates good practises, hence why enforce it with a bit of a hack.